Project 4_KTs

```diff
@@ -0,0 +1,446 @@
+
+Test Cases:
+
+The query should be tested on a large RDF dataset that contains a variety of scientific publications, authors, and citation links. The dataset should also contain some invalid or incomplete data to ensure that the query can handle such scenarios.
+
+Here are some example test cases:
+
+The query should return the correct number of authors and their publications.
+The query should correctly identify the citation links between publications and authors.
+The visualization should accurately represent the citation network and allow the user to interact with the graph to explore the data.
+The query should be efficient and able to handle large datasets in a reasonable amount of time.
+
+The following is a general outline of the steps that could be taken to solve this kata:
+
+Identify the top 20 authors with the highest number of publications using the COUNT() and ORDER BY functions in SPARQL.
+Retrieve the publications and authors cited by each top author using the SELECT, WHERE, and FILTER clauses in SPARQL.
+Iterate over the cited authors and retrieve their publications and authors cited until a maximum depth of 3 is reached. This can be achieved using SPARQL subqueries or recursive queries.
+Transform the resulting data into a graph format that can be visualized using a graph visualization tool such as D3.js or NetworkX.
+Visualize the graph and allow the user to interact with it to explore the citation network.
+The query will need to be tested and refined to ensure that it correctly handles invalid or incomplete data and can efficiently handle large datasets.
+
+Proposed answer to SPARQL query:
+
+---
+PREFIX rdf: <http://www.w3.org/1999/02/22-rdf-syntax-ns#>
+PREFIX rdfs: <http://www.w3.org/2000/01/rdf-schema#>
+
+# Identify the top 20 authors with the highest number of publications
+SELECT ?author (COUNT(?pub) AS ?numPublications)
+WHERE {
+  ?pub rdf:type ?type .
+  ?type rdfs:subClassOf* <http://purl.org/spar/fabio/Expression> .
+  ?pub <http://purl.org/dc/elements/1.1/creator> ?author .
+}
+GROUP BY ?author
+ORDER BY DESC(?numPublications)
+LIMIT 20
+
+# Retrieve the publications and authors cited by each top author
+SELECT ?author ?pub ?citedAuthor
+WHERE {
+  ?pub rdf:type ?type .
+  ?type rdfs:subClassOf* <http://purl.org/spar/fabio/Expression> .
+  ?pub <http://purl.org/dc/elements/1.1/creator> ?author .
+  ?pub <http://purl.org/spar/fabio/cites> ?cited .
+  ?cited <http://purl.org/dc/elements/1.1/creator> ?citedAuthor .
+}
+ORDER BY ?author
+
+# Retrieve the publications and authors cited by each cited author
+SELECT ?citedAuthor ?pub ?citedAuthor2
+WHERE {
+  ?pub rdf:type ?type .
+  ?type rdfs:subClassOf* <http://purl.org/spar/fabio/Expression> .
+  ?pub <http://purl.org/dc/elements/1.1/creator> ?citedAuthor .
+  ?pub <http://purl.org/spar/fabio/cites> ?cited .
+  ?cited <http://purl.org/dc/elements/1.1/creator> ?citedAuthor2 .
+}
+ORDER BY ?citedAuthor
+---
```

Comment:
I love this. This is a much more complicated problem and solution.

I think this should be a Kata-4 maybe. See my prior comments on what I would expect for a Kata at level 4. 

I think some of this could be cleaned up, since I am not sure why there are two SELECT queries. Is there a way to put them together, in UNION or to make one select nested in another? This might make it a solid Kata-4, or, depending on the conceptual difficulties, a Kata-3.